### ISSUE TITLE
Unable to using other languages except english for jhi-item-count.component.ts labels.

### ISSUE THREAD

**Issue by devanonym** [jhipster/generator-jhipster#5739]
##### **Overview of the issue**
 Jhi-item-count component of ng-jhipster library contains hardcoded (english) values of labels  - Showing, of , items

##### **Motivation for or Use Case**

Simple motivation -  use another languages for a labels.

##### **Reproduce the error**
 Generate a project with pagination option.

##### **JHipster Version(s)**
4.4.1

**Comment by JulienMrgrd** [comment]
I'm on it :)

**Comment by JulienMrgrd** [comment]
I'll do 2 PR separately.
One in the `generator-jhipster` (a new key in all i18n files) 
and another in `ng-jhipster` (the item-count component).

I block slightly on a test issue in ng-jhipster, but the fix comes up ;)

Concerning the new key, I only managed to translate (more or less correctly) :

- pt-br
- pt
- fr
- de
- sv
- cs
- da
- el
- ca
- it
- es
- nl
- ro
- gl

(Thanks Google Translate !)

**Comment by gzsombor** [comment]
The tricky part is, that we need a way to generate an application, without i18n support. So either we keep the old, non-i18n component in ng-jhipster, and in generation-time we pick the appropriate one, or the component needs to be enhanced to support the two way of operations

**Comment by deepu105** [comment]
We can use the conditions passed during the ng-jhipster module
initialization to handle i18n and non i18n. Like how its done for the alert
componenets


Thanks & Regards,
Deepu

**Comment by JulienMrgrd** [comment]
I'm actually coding a way using both a @HostBinding and an @Input, on the jhi-item component. 
Something like this (can maybe change a little)  :

```
@Input() i18nSupported: boolean;

@HostBinding('attr.jhiTranslate') jhiTranslate: string = !this.i18nSupported ? null : 'global.item-count';

@HostBinding('attr.translateValues') translateValues: string = !this.i18nSupported ? null :
        '{first: \'+{{((page - 1) * itemsPerPage) == 0 ? 1 : ((page - 1) * itemsPerPage + 1)}}\',' +
        'second: \'{{(page * itemsPerPage) < total ? (page * itemsPerPage) : total}}\',' +
        'total: \'{{total}}\'}';
```

In this way, when initialization, we only have to add a `i18nSupported="true"` to the jhi-item directive, an the translation attribute will be added.

What do you prefer ?

**Comment by deepu105** [comment]
Can you look at how its handled in aler components i would like behaviour
to be same on all

**Comment by JulienMrgrd** [comment]
That's ok

**Comment by JulienMrgrd** [comment]
Since we can not use the same way the _Alert service_ and the _jhi-item-count component_, I propose to use what @deepu105 had indicated, concerning the 18n conditions.

In ng-jhipster, by calling `new ModuleConfig().i18nEnabled`, the jhi-item-count component can know if it should leave the default template, or if it should add the "jhiTranslate" and "translateValues" attributes (see in **jhi-item-count.component.ts**).

As a result, no modification needed in the generator, just a few lines (which I have already coded) on ng-jhipster, and some minor modifications of the tests.

(and of course, there is no longer need what I suggested yesterday, concerning a new @Input "i18nSupported" or @HostBindings).
